pimd/pim_upstream.c

```diff
@@ -1713,10 +1713,16 @@ int pim_upstream_inherited_olist_decide(struct pim_instance *pim,
 
 		if (pim_upstream_evaluate_join_desired_interface(up, ch,
 								 starch)) {
-			int flag = PIM_OIF_FLAG_PROTO_PIM;
+			int flag = 0;
```

Comment:
enough to declare with `int flag;`

Reply:
nah not all paths below set flag as such we should give it an initial value that makes sense.

Reply:
flag=0 initialisation is required because in the else condition, if both condition are not true.